src/main/java/org/jenkinsci/plugins/github/config/GitHubPluginConfig.java

```diff
@@ -76,6 +78,7 @@
     @SuppressWarnings("unused")
     private transient InstanceIdentity identity;
 
+    @DataBoundConstructor
     public GitHubPluginConfig() {
         load();
```

Comment:
DataBoundConstructor is used for binding new data, `load()` is loading old data. Thats looks wrong

Reply:
This is the constructor being used by CasC, and it uses the `getter` and `setter` to mutate the instance. The other constructor would make `List<GitHubServerConfig> configs` required, which should not. What do you suggest would be an appropriate fix to address your concerns?

Reply:
@jglick could you describe why DBC is set on top of Bean constructor that loads data?

Reply:
This is wrong; the annotation should be deleted. `GitHubPluginConfig` is a singleton (`GlobalConfiguration`) so it may have `@DataBoundSetter`s but that is it.